src/Particles/babylon.particleSystem.ts

```diff
@@ -555,10 +571,139 @@
             this._vertexData[offset + 11] = particle.cellIndex;
         }
 
+        // start of sub system methods
+
+        /**
+         * "Recycles" one of the particle by copying it back to the "stock" of particles and removing it from the active list.
+         * Its lifetime will start back at 0.
+         */
+        public recycleParticle: (particle: Particle) => void = (particle) => {
+            this._recycleParticleUsingSystem(this, this, particle);
+        };
+
+        private _stopSubEmitters(): void {
+            this.stockSubSystems.forEach(index => {
+                this.stockSubSystems.get(index)!.forEach(subSystem => {
+                    subSystem.stop();
+                });
+            });
+
+            this.activeSubSystems.forEach(subSystem => {
+                subSystem.stop();
+                subSystem._stoppedEmitting();
+            });
+            this.activeSubSystems = new Array<ParticleSystem>();
+        }
+
+        private _createParticle: () => Particle = () => {
+            return this._createParticleUsingSystem(this, this);
+        }
+
+        // to be overriden by subSystems
+        private _stoppedEmitting: () => void = () => {

```

Comment:
why this?

Reply:
I still need to dispose the last generated ParticleSystem because it was dropping FPS in a huge way :) after 10 mins to was dropped to 30.

Reply:
I will merge the main and sub into one function